From 5ddc2e81007b73988c7c5ca2580e8cc579efa069 Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Thu, 12 Jul 2007 03:08:37 +0000
Subject: [PATCH] Add test for BEANUTILS-140 - LocaleBeanUtils setProperty does
 not work on nested property - reposted by Marco La Porta

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/beanutils/trunk@555476 13f79535-47bb-0310-9956-ffa450edef68
---
 .../locale/LocaleBeanUtilsTestCase.java       | 106 ++++++++++++++++++
 1 file changed, 106 insertions(+)
 create mode 100644 src/test/org/apache/commons/beanutils/locale/LocaleBeanUtilsTestCase.java

diff --git a/src/test/org/apache/commons/beanutils/locale/LocaleBeanUtilsTestCase.java b/src/test/org/apache/commons/beanutils/locale/LocaleBeanUtilsTestCase.java
new file mode 100644
index 000000000..76a8ffc12
--- /dev/null
+++ b/src/test/org/apache/commons/beanutils/locale/LocaleBeanUtilsTestCase.java
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.beanutils.locale;
+
+import junit.framework.TestCase;
+import junit.framework.Test;
+import junit.framework.TestSuite;
+
+import org.apache.commons.beanutils.TestBean;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Test Case for {@link LocaleBeanUtils}.
+ *
+ * @version $Revision$ $Date$
+ */
+public class LocaleBeanUtilsTestCase extends TestCase {
+    
+    private static Log log = LogFactory.getLog(LocaleBeanUtilsTestCase.class);
+    
+    /**
+     * Construct a new instance of this test case.
+     *
+     * @param name Name of the test case
+     */
+    public LocaleBeanUtilsTestCase(String name) {
+        super(name);
+    }
+
+
+    // -------------------------------------------------- Overall Test Methods
+
+
+    /**
+     * Set up instance variables required by this test case.
+     */
+    public void setUp() {
+    }
+
+
+    /**
+     * Return the tests included in this test suite.
+     * @return Test Suite
+     */
+    public static Test suite() {
+        return (new TestSuite(LocaleBeanUtilsTestCase.class));
+    }
+
+
+    /**
+     * Tear down instance variables required by this test case.
+     */
+    public void tearDown() {
+    }
+
+
+    // ------------------------------------------------ Individual Test Methods
+
+    /**
+     * Test setting a nested simple property
+     */
+    public void testSetNestedPropertySimple() {
+        TestBean bean = new TestBean();
+        bean.getNested().setIntProperty(5);
+        assertEquals("Initial value 5", 5, bean.getNested().getIntProperty());
+        try {
+            LocaleBeanUtils.setProperty(bean, "nested.intProperty", "123", null);
+        } catch (Throwable t) {
+            log.error(t);
+            fail("Threw " + t);
+        }
+        assertEquals("Check Set Value", 123, bean.getNested().getIntProperty());
+    }
+
+    /**
+     * Test setting a nested indexed property
+     */
+    public void testSetNestedPropertyIndexed() {
+        TestBean bean = new TestBean();
+        bean.getNested().setIntIndexed(1, 51);
+        assertEquals("Initial value[1] 51", 51, bean.getNested().getIntIndexed(1));
+        try {
+            LocaleBeanUtils.setProperty(bean, "nested.intIndexed[1]", "123", null);
+        } catch (Throwable t) {
+            log.error(t);
+            fail("Threw " + t);
+        }
+        assertEquals("Check Set Value", 123, bean.getNested().getIntIndexed(1));
+    }
+}
+
